public/js/main.js

```diff
@@ -1,113 +1,177 @@
+            //Conditional script here
+            document.addEventListener('fullscreenchange', toggleFullScreen, false);
+            document.addEventListener('mozfullscreenchange', toggleFullScreen, false);
+            document.addEventListener('MSFullscreenChange', toggleFullScreen, false);
+            document.addEventListener('webkitfullscreenchange', toggleFullScreen, false);
+        }
     } else {
-        window.onresize = setSize();
+        log("Could not complete setup as DOM not loaded", "error");
     }
 
-    document.addEventListener('fullscreenchange', toggleFullScreen, false);
-    document.addEventListener('mozfullscreenchange', toggleFullScreen, false);
-    document.addEventListener('MSFullscreenChange', toggleFullScreen, false);
-    document.addEventListener('webkitfullscreenchange', toggleFullScreen, false);
+    document.onkeypress = keyboardEvent;
 }
 
-function start_handler() {
-    toggleFullScreen(true);
+export function keyboardEvent(event) {  // spacebar
+    if (event.which == "32") engine.processJump();
+}
+
+export function startHandler() {
+    // if the user has a mobile device, allow fullscreen. Otherwise it will impact the user experience, so disable it.
+    if (window.matchMedia("only screen and (max-width: 768px)").matches === true || window.matchMedia("only screen and (max-height: 768px)").matches === true) { 
+        toggleFullScreen(true);
+    }
+
+    document.querySelector("#idle_background").style.display = 'none';
+    document.querySelector("#playing_background").style.display = 'block';
     document.querySelector("#runner_container").style.display = "block";
     document.querySelector("#runner_before").style.display = "none";
-    setSize(); //make sure canvas is sized properly.
+    setSize(ctx); //make sure canvas is sized properly.
     then = Date.now();
     run(); //start the animation loop.
 }
 
-function setSize() {
-    let original_size = [ctx.canvas.width, ctx.canvas.height];
-    ctx.canvas.width = window.innerWidth;
-    fpsInterval = 30 - (ctx.canvas.width / 250); 
+export function setSize(context) {
+    if (!context.width) context = ctx; // for testing
+    let original_size = [context.canvas.width, context.canvas.height];
+    context.canvas.width = window.innerWidth;
+    fpsInterval = Math.floor(30 - (context.canvas.width / 250)); 
 
-    const last_element = document.querySelectorAll("#Runner h3")[1];
-    ctx.canvas.height = window.innerHeight - (last_element.offsetHeight + last_element.getBoundingClientRect().bottom);
+    context.canvas.height = window.innerHeight;
     if (engine) {
         // Increase the fps for higher pixel counts to prevent ghosting.
         // Higher pixel counts also necessitate faster CPUs and GPUs, so a higher framerate is more tolerable.
-        engine.adjust_for_fps(1000/fpsInterval);
+        engine.adjustForFps(1000/fpsInterval);
```

Comment:
Evidence throughout to fulfill requirement 1: consistent naming scheme (camelCase).